test/FastTests/Server/Documents/Indexing/Staggered/StaggeredIndexesClusterTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Indexes.Spatial;
+using Raven.Client.Documents.Operations;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Operations;
+using Raven.Server.Config;
+using Raven.Server.Json;
+using Raven.Server.Documents.Indexes;
+using Raven.Server.Documents.Indexes.Static;
+using Raven.Server.Documents.Queries;
+using Raven.Server.Extensions;
+using Raven.Server.ServerWide;
+using Raven.Server.Utils;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace FastTests.Server.Documents.Indexing.Static
+{
+    public class StaggeredIndexesClusterTests : ClusterTestBase
+    {
+        public StaggeredIndexesClusterTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task DeployStaticStaggeredIndex()
+        {
+            DebuggerAttachedTimeout.DisableLongTimespan = false;
+            var cluster = await CreateRaftCluster(3, watcherCluster: true);
+            using (var leaderStore = GetDocumentStore(new Options
+            {
+                Server = cluster.Leader,
+                ReplicationFactor = 3
+            }))
+            {
+                var dbName = leaderStore.Database;
+
+                await leaderStore.Maintenance.SendAsync(new CreateSampleDataOperation());
+
+                await leaderStore.Operations.SendAsync(new PatchByQueryOperation(@"FROM Orders UPDATE { for(var i=0; i<100; i++) put(""Orders/"", this); }"));
+
+                var indexDefinition = new IndexDefinition
+                {
+                    Name = "order_companies",
+                    Maps = { "from order in docs.Orders select new { company = order.Company }" },
+                    Staggered = true
+                };
+
+                var db = await cluster.Leader.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(dbName);
```

Comment:
You aren't actually testing this?

Reply:
No, this is just a draft and it's not ready yet.
I just wanted to see if the general approach is fine before I write proper tests and error handling.

Reply:
Understood, and perfectly fine for draft PR / review / in progress.

Testing this might mean that we insert some hooks to enable that, though.